core/src/main/java/org/elasticsearch/cluster/ClusterInfo.java

```diff
@@ -68,25 +68,25 @@ public ClusterInfo(ImmutableOpenMap<String, DiskUsage> leastAvailableSpaceUsage,
     }
 
     public ClusterInfo(StreamInput in) throws IOException {
-        int size = in.readInt();
+        int size = in.readVInt();
```

Comment:
I wonder if we should introduce a method to InputStream/OutputStream to read maps. We have similar methods for lists. Something like:
```
<K, V> Map<K, V> readMap(Reader<K> keyReader, Reader<V> valueReader)
```

I can think of about a dozen places we could use it.